src/main/java/org/skriptlang/skript/bukkit/tags/TagRegistry.java

```diff
@@ -0,0 +1,206 @@
+package org.skriptlang.skript.bukkit.tags;
+
+import ch.njol.util.coll.iterator.CheckedIterator;
+import com.destroystokyo.paper.MaterialTags;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Iterators;
+import io.papermc.paper.tag.EntityTags;
+import org.bukkit.Keyed;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Tag;
+import org.bukkit.entity.EntityType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.tags.sources.BukkitTagSource;
+import org.skriptlang.skript.bukkit.tags.sources.PaperTagSource;
+import org.skriptlang.skript.bukkit.tags.sources.SkriptTagSource;
+import org.skriptlang.skript.bukkit.tags.sources.TagOrigin;
+import org.skriptlang.skript.bukkit.tags.sources.TagSource;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * A class in charge of storing and handling all the tags Skript can access.
+ */
+public class TagRegistry {
+
+	private final TagSourceMap tagSourceMap = new TagSourceMap();
+
+	/**
+	 * Each new instance will create a new set of tag sources, in an effort to be reload safe.
```

Comment:
This is not true for Skript tags, is it? Should it be?

Reply:
It probably should be yeah